trajectory_processing/src/iterative_time_parameterization.cpp

```diff
@@ -111,13 +112,18 @@ void IterativeParabolicTimeParameterization::applyVelocityConstraints(robot_traj
   {
     const robot_state::RobotStatePtr &curr_waypoint = rob_trajectory.getWayPointPtr(i);
     const robot_state::RobotStatePtr &next_waypoint = rob_trajectory.getWayPointPtr(i+1);
-    
+
+    double velocity_scaling_factor = 1.0;
+
+    if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+      velocity_scaling_factor = max_velocity_scaling_factor;
```

Comment:
Updated, see https://github.com/team-vigir/moveit_core/commit/58d1660e8bb540042e47a4168ad9221d5940f392. Now sets to 1.0 when 0.0 is incoming with logDebug output and provides logWarn output if specified factor is outside [0.0, 1.0] (also using 1.0 default instead).